### DISCUSSION TITLE
How to store an object (WebSocket) globally?

### DISCUSSION THREAD

**Discussion by arun-mani-j** [nuxt/nuxt#25082] [Questions] [answered]
Hii, this might be a repeated question, but I'm asking I'm not able to find a meaningful solution.

My app is fully client-side rendered. It needs to create a WebSocket. This socket has to be shared across all the pages and components. What is the best way to do this?

I have found `useNuxtApp` composable (https://nuxt.com/docs/api/composables/use-nuxt-app) and it looks to be what I need. Is that right? Thanks in advance!

**Comment by TheAlexLichter** [comment, marked as answer]
provide it in a plugin, then use it via `useNuxtApp` indeed. Another option is a composable, e.g. based on https://vueuse.org/core/useWebSocket/
